packages/workflow-angular/lib/workflow-process/workflow-process-steps/workflow-process-steps-controller.js

```diff
@@ -46,18 +45,18 @@ function WorkflowProcessStepsController($scope, $state, mediator, workflowMediat
   }
 
   self.back = function() {
-    workflowMediatorService.previousStep(workorderId).then(function(workflowSummary) {
+    workflowApiService.previousStep(workorderId).then(function(workflowSummary) {
       updateWorkflowState(workflowSummary);
     });
   };
 
   //Beginning the workflow
-  workflowMediatorService.beginWorkflow(workorderId).then(function(workflowSummary) {
+  workflowApiService.beginWorkflow(workorderId).then(function(workflowSummary) {
     updateWorkflowState(workflowSummary);
   });
 
-  mediator.subscribeForScope("wfm:workflow:step:done", $scope, function(submission) {
-    workflowMediatorService.completeStep({
+  workflowApiService.nextStepSubscriber(function(submission) {
```

Comment:
Maybe rename these to verbs, nouns as functions is a bit weird. i.e. `subscribeToNextStep()`

I know you don't like `EventEmmiter`, but it takes care of some things like multiple subscribers and such